bigtable/google/cloud/bigtable/row_set.py

```diff
@@ -29,6 +29,32 @@ def __init__(self):
         self.row_keys = []
         self.row_ranges = []
 
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+
+        if len(other.row_keys) != len(self.row_keys):
+            return False
+
+        if len(other.row_ranges) != len(self.row_ranges):
+            return False
+
+        if not sorted(other.row_keys) == sorted(self.row_keys):
+            return False
+
+        row_ranges = list(self.row_ranges)
+
+        for each in other.row_ranges:
+            try:
+                row_ranges.remove(each)
+            except ValueError:
+                return False
```

Comment:
This seems more complicated than necessary, and is likely slow (raising / catching exceptions is lots slower than normal code).  I think we can use sets here as well:

```python
        if not set(other.row_ranges) == set(self.row_ranges):
            return False
```

Reply:
Added __hash__() to make RowRange hashable. Now, we can use set while comparing row_ranges.